### ISSUE TITLE
Stop/Reload methods simplification + harden SIGINT handling 

### ISSUE THREAD

**Issue by Unitech** [Unitech/pm2#2481]
#2333 Pm2 reload doesn't send SIGINT signal on cluster mode
#2478 Downtime when using reload and wait_ready
#1732 Graceful Exit
#2025 What's the difference between **reload** and **gracefulReload**?
#1346 graceful scale down
#1763 graceful shutdown not working
#1234 Is it possible to require process.send("online") before it goes online?
#856 pm2 doesn't log after shutdown
#1166 reload and gracefulReload kill processes too quickly (not 0s downtime)
#1637 amqp gracefulRload Assertion
#1311 PM2 clustered mode laod balancing issue
#1101 Errors when using gracefulReload (or reload)

Available in:

```
$ npm install pm2@next -g
```

**Comment by Digot** [comment]
Hey, do you know when you will be able to merge the stop-refactor branch into master? (Or how I can run pm2 with these changes already?). 

Thanks!

**Comment by Unitech** [comment]
This is now available on @next tag:

```bash
$ npm install pm2@next -g
```

Feedback appreciated

**Comment by Unitech** [comment]
Patch available on pm2@2.1.4 (main):

```bash
$ npm install pm2 -g
$ pm2 update
```
